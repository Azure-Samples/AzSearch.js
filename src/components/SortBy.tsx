import * as React from "react";
import { PropsType } from "../containers/SortByContainer";
import * as objAssign from "object-assign";
import { Store } from "azsearchstore";
import { defaultCss } from "../utils/css";

export type State = {};

class SortBy extends React.PureComponent<PropsType, State> {
  render() {
    const { fields, beforeFirstRequest, onSortChange, orderby } = this.props;
    let css = objAssign({}, defaultCss, this.props.css);
    let order = "desc";
    if (beforeFirstRequest) return <div></div>;

    let options = fields.map((field, index) => {
      return <option  key={index} selected={field.orderbyClause === orderby} value={field.orderbyClause}>{field.displayName}</option>;
    });

    let onChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
      onSortChange(event.target.value);
    };

    let container = <div className={css.sorting__sortBy}>
      <div className={css.searchFacets__facetHeaderContainer}>
          <h4 className={css.searchFacets__facetHeader}>
              <a data-toggle="collapse" className={css.searchFacets__facetHeaderLink}  >
                  <span className={css.searchFacets__facetHeaderIconOpen} aria-hidden="true"></span> Sort By
              </a>
          </h4>
      </div>
      <div className={css.searchFacets__facetControlContainer}>
        <div className={css.searchFacets__facetControlList}>
          <select className={css.sorting__sortByControl} onChange={onChange}>
            {options}
          </select>
        </div>
      </div>
    </div>;

    return (container);
  }
}

export default SortBy;