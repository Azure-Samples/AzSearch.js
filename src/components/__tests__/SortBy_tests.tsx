import * as React from "react";
import { shallow } from "enzyme";
import SortBy from "../SortBy";

const fields = [
    { displayName: "Relevance", orderbyClause: "" },
    { displayName: "Beds", orderbyClause: "beds desc" },
    { displayName: "Baths", orderbyClause: "baths desc" }
];
const orderby = "";

describe("<SortBy/>", () => {
    it("renders against dummy snapshot", () => {

        expect(shallow(
            <SortBy  beforeFirstRequest={false} onSortChange={function(foo){}} fields={fields} css={null} orderby={orderby} />
        )).toMatchSnapshot();
    });
});