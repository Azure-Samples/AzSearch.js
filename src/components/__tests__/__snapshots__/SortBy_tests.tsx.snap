// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortBy/> renders against dummy snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="sorting__sortyBy panel-body"
>
    <div
        className="searchResults__facetHeader-container panel-heading"
    >
        <h4
            className="searchResults__facetHeader panel-title"
        >
            <a
                className="searchResults__facetHeader-link"
                data-toggle="collapse"
            >
                <span
                    aria-hidden="true"
                    className="searchResults__facetHeader-icon--open indicator glyphicon glyphicon-triangle-bottom"
                />
                 Sort By
            </a>
        </h4>
    </div>
    <div
        className="searchResults__facetControl-container panel-collapse collapse in"
    >
        <div
            className="searchResults__facetControl-list list-group"
        >
            <select
                className="sorting__sortyByControl list-group-item"
                onChange={[Function]}
            >
                <option
                    selected={true}
                    value=""
                >
                    Relevance
                </option>
                <option
                    selected={false}
                    value="beds desc"
                >
                    Beds
                </option>
                <option
                    selected={false}
                    value="baths desc"
                >
                    Baths
                </option>
            </select>
        </div>
    </div>
</div>,
  "nodes": Array [
    <div
      className="sorting__sortyBy panel-body"
>
      <div
            className="searchResults__facetHeader-container panel-heading"
      >
            <h4
                  className="searchResults__facetHeader panel-title"
            >
                  <a
                        className="searchResults__facetHeader-link"
                        data-toggle="collapse"
                  >
                        <span
                              aria-hidden="true"
                              className="searchResults__facetHeader-icon--open indicator glyphicon glyphicon-triangle-bottom"
                        />
                         Sort By
                  </a>
            </h4>
      </div>
      <div
            className="searchResults__facetControl-container panel-collapse collapse in"
      >
            <div
                  className="searchResults__facetControl-list list-group"
            >
                  <select
                        className="sorting__sortyByControl list-group-item"
                        onChange={[Function]}
                  >
                        <option
                              selected={true}
                              value=""
                        >
                              Relevance
                        </option>
                        <option
                              selected={false}
                              value="beds desc"
                        >
                              Beds
                        </option>
                        <option
                              selected={false}
                              value="baths desc"
                        >
                              Baths
                        </option>
                  </select>
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <SortBy
        beforeFirstRequest={false}
        css={null}
        fields={
                Array [
                        Object {
                          "displayName": "Relevance",
                          "orderbyClause": "",
                        },
                        Object {
                          "displayName": "Beds",
                          "orderbyClause": "beds desc",
                        },
                        Object {
                          "displayName": "Baths",
                          "orderbyClause": "baths desc",
                        },
                      ]
        }
        onSortChange={[Function]}
        orderby=""
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SortBy {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "beforeFirstRequest": false,
          "css": null,
          "fields": Array [
            Object {
              "displayName": "Relevance",
              "orderbyClause": "",
            },
            Object {
              "displayName": "Beds",
              "orderbyClause": "beds desc",
            },
            Object {
              "displayName": "Baths",
              "orderbyClause": "baths desc",
            },
          ],
          "onSortChange": [Function],
          "orderby": "",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="sorting__sortyBy panel-body"
>
          <div
                    className="searchResults__facetHeader-container panel-heading"
          >
                    <h4
                              className="searchResults__facetHeader panel-title"
                    >
                              <a
                                        className="searchResults__facetHeader-link"
                                        data-toggle="collapse"
                              >
                                        <span
                                                  aria-hidden="true"
                                                  className="searchResults__facetHeader-icon--open indicator glyphicon glyphicon-triangle-bottom"
                                        />
                                         Sort By
                              </a>
                    </h4>
          </div>
          <div
                    className="searchResults__facetControl-container panel-collapse collapse in"
          >
                    <div
                              className="searchResults__facetControl-list list-group"
                    >
                              <select
                                        className="sorting__sortyByControl list-group-item"
                                        onChange={[Function]}
                              >
                                        <option
                                                  selected={true}
                                                  value=""
                                        >
                                                  Relevance
                                        </option>
                                        <option
                                                  selected={false}
                                                  value="beds desc"
                                        >
                                                  Beds
                                        </option>
                                        <option
                                                  selected={false}
                                                  value="baths desc"
                                        >
                                                  Baths
                                        </option>
                              </select>
                    </div>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="sorting__sortyBy panel-body"
>
          <div
                    className="searchResults__facetHeader-container panel-heading"
          >
                    <h4
                              className="searchResults__facetHeader panel-title"
                    >
                              <a
                                        className="searchResults__facetHeader-link"
                                        data-toggle="collapse"
                              >
                                        <span
                                                  aria-hidden="true"
                                                  className="searchResults__facetHeader-icon--open indicator glyphicon glyphicon-triangle-bottom"
                                        />
                                         Sort By
                              </a>
                    </h4>
          </div>
          <div
                    className="searchResults__facetControl-container panel-collapse collapse in"
          >
                    <div
                              className="searchResults__facetControl-list list-group"
                    >
                              <select
                                        className="sorting__sortyByControl list-group-item"
                                        onChange={[Function]}
                              >
                                        <option
                                                  selected={true}
                                                  value=""
                                        >
                                                  Relevance
                                        </option>
                                        <option
                                                  selected={false}
                                                  value="beds desc"
                                        >
                                                  Beds
                                        </option>
                                        <option
                                                  selected={false}
                                                  value="baths desc"
                                        >
                                                  Baths
                                        </option>
                              </select>
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SortBy
    beforeFirstRequest={false}
    css={null}
    fields={
        Array [
            Object {
              "displayName": "Relevance",
              "orderbyClause": "",
            },
            Object {
              "displayName": "Beds",
              "orderbyClause": "beds desc",
            },
            Object {
              "displayName": "Baths",
              "orderbyClause": "baths desc",
            },
          ]
    }
    onSortChange={[Function]}
    orderby=""
/>,
}
`;
